src/tests/tracing/eventlistener/eventlistenerenabledisable.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Diagnostics.Tracing;
+using System.Threading;
+using Tracing.Tests.Common;
+
+namespace Tracing.Tests
+{
+    [EventSource(Name = "Tracing.Tests.EnableDisableEventSource")]
+    class EnableDisableEventSource : EventSource
+    {
+        internal int _enables;
+        internal int _disables;
+
+        public EnableDisableEventSource() : base(true) { }
+
+        [Event(1)]
+        internal void TestEvent() { this.WriteEvent(1); }
+
+        [NonEvent]
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            base.OnEventCommand(command);
+
+            if (command.Command == EventCommand.Enable)
+            {
+                Interlocked.Increment(ref _enables);
+            }
+            else if (command.Command == EventCommand.Disable)
+            {
+                Interlocked.Increment(ref _disables);
+            }
+            else
+            {
+                throw new Exception($"Saw unexpected command {command.Command} in OnEventCommand");
+            }
+        }
+    }
+    
+    internal sealed class EnableDisableListener : EventListener
+    {
+        private EventSource? _target;
+
+        protected override void OnEventSourceCreated(EventSource source)
+        {
+            if (source.Name.Equals("Tracing.Tests.EnableDisableEventSource"))
+            {
+                _target = source;
+                EnableEvents(_target, EventLevel.Verbose);
+            }
+        }
+
+        public void Disable()
+        {
+            DisableEvents(_target);
+        }
+
+        public void Dispose(bool disableEvents)
+        {
+            base.Dispose();
+
+            if (disableEvents)
+            {
+                // Dispose should call DisableEvents for us, this should be ignored after Dispose()
+                Disable();
+            }
+        }
+    }
+
+    class EventPipeSmoke
+    {
+        private static int messageIterations = 100;
+        private static readonly DateTime ThePast = DateTime.UtcNow;
```

Comment:
Are these fields used anywhere?

Reply:
No, good catch. Just copy and paste leftovers